frontend/src/Menu.js

```diff
@@ -1,19 +1,17 @@
-import React from 'react'
-import { NavLink } from 'react-router-dom'
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+import GetDecodedToken from './GetDecodedToken';
 
 function Menu() {
+  const isAdmin = GetDecodedToken()?.resource_access.worksheetclient.roles.includes('admin');
```

Comment:
Ezt a logikát is lehetne a hook-on belülre rejteni, mondjuk a hook neve ebben az esetben lehetne más: useCurrentUser és az isAdmin-t ő állapítaná meg és adná vissza. A token sok mindent tartalmazhat még, elég lenne csak azokat az adatokat visszaadni, amire épít az alkalmazás.